docs/proposals/configuration_changes.md

```diff
@@ -0,0 +1,71 @@
+## Configuration Changes
+
+### Goal
+Create a package for creating and managing configuration values as a value store. This will be used to remove the necessity of each package to be loaded just for configuration values. The other benefit is that this package could be a more robust solution to the configuration, Allowing us to do a better job of watching for updates to the configmap and managing subscriptions to those changes. This value store should also be usable by other parts of the application and therefore will require less usage of hard coded configuration structures and will allow the addition or removal of configuration values to not create such an issue. 
```

Comment:
Is there a use case for config changes changing while a broker is running? Right now we just update the broker-config and bounce the pod.

Reply:
I think that we should eventually do this. I think it is a nice to have feature down the road, but that we should eventually have it. I don't see why, when I update the config map I then have to bounce the pod as the ideal end state IMO. I am not saying that as part of the initial PR to move to the new configuration that needs to work, just mentioning that moving to this new way will enable this feature in the future.

Does that make any sense?

Reply:
So you want to edit the config file and have that trigger something in the broker that reconfigures it? I guess I just don't see this being used often, unless I'm misunderstanding.